pkg/errors/broadcasting.go

```diff
@@ -0,0 +1,213 @@
+package errors
+
+import (
+	"errors"
+	"fmt"
+	"log/slog"
+
+	"github.com/bsv-blockchain/go-sdk/transaction"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/wdk"
+)
+
+// BroadcastOperation defines the type for various operations in the wallet system.
+type BroadcastOperation string
+
+const (
+	// BackgroundBroadcast represents a background broadcasting operation.
+	BackgroundBroadcast BroadcastOperation = "backgroundBroadcast"
+
+	// ImmediateBroadcast represents an immediate broadcasting operation.
+	ImmediateBroadcast BroadcastOperation = "immediateBroadcast"
+
+	// DelayedBroadcast represents a delayed broadcasting operation.
+	DelayedBroadcast BroadcastOperation = "delayedBroadcast"
+
+	// CreateAction represents an action to create a new wallet action.
+	CreateAction BroadcastOperation = "createAction"
+
+	// ProcessAction represents an action to process an existing wallet action.
+	ProcessAction BroadcastOperation = "processAction"
+)
+
+// BroadcastingError represents an error that occurred during transaction broadcasting
+type BroadcastingError struct {
+	Err             error
+	TxID            string
+	Reference       string
+	SendWithResults []wdk.SendWithResult
+	ReviewResults   []wdk.ReviewActionResult
+	ServiceErrors   map[string]error
+	Operation       BroadcastOperation
+	Tx              []byte
+	NoSendChange    []wdk.OutPoint
+}
+
+// Error implements the error interface for BroadcastingError
+func (e *BroadcastingError) Error() string {
+	if e.Err == nil {
+		return fmt.Sprintf("broadcasting failed during %s for txID %s (reference: %s)",
```

Comment:
I think that we loose a lot of context here, that it was stored on BroadcastingError